bot/internal/bot/backport.go

```diff
@@ -182,32 +213,23 @@ func (b *Bot) createBackportBranch(ctx context.Context, organization string, rep
 	}
 
 	// Download base and head from origin (GitHub).
-	log.Println("Running:", "git fetch origin", base, pull.UnsafeHead.Ref)
 	if err := git("fetch", "origin", base, pull.UnsafeHead.Ref); err != nil {
 		return trace.Wrap(err)
 	}
 
-	// Checkout the base branch then rebase commits from Pull Request ontop of
-	// it. See https://stackoverflow.com/a/29916361 for more details.
-	newParent := base
-	oldParent := pull.UnsafeBase.SHA
-	until := pull.UnsafeHead.SHA
-	if err := git("checkout", base); err != nil {
+	// Checkout the new backport branch.
+	if err := git("checkout", "-b", newHead); err != nil {
 		return trace.Wrap(err)
 	}
 
-	log.Println("Running: git rebase --onto", newParent, oldParent, until)
-	if err := git("rebase", "--onto", newParent, oldParent, until); err != nil {
-		if er := git("rebase", "--abort"); er != nil {
-			return trace.NewAggregate(err, er)
+	// Cherry-pick all commits from the PR to the backport branch.
+	for _, commit := range pull.Commits {
```

Comment:
Can we just look at `pull.MergeCommitSHA` instead of `pull.Commits`? We've enabled squash and merge for the merge queue, so there should only ever be one squashed commit getting merged.

Reply:
I thought about this but hopefully we'll eventually get back the ability to optionally use rebase & merge at some point and I'd rather avoid having to make another fix to the bot then. This approach should work for any merge strategy.